src/Symfony/Component/Utf8/Bytes.php

```diff
@@ -0,0 +1,505 @@
+    {
+        if ('' === $needle) {
+            return;
+        }
+
+        $current = setlocale(LC_CTYPE, '0');
+        setlocale(LC_CTYPE, 'C');
+        $part = stristr($this->string, $needle, $beforeNeedle);
+        setlocale(LC_CTYPE, $current);
+
+        if (false === $part) {
+            return;
+        }
+
+        $result = clone $this;
+        $result->string = $part;
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lastSubstringOf(string $needle, bool $beforeNeedle = false)
+    {
+        if ('' === $needle) {
+            return;
+        }
+
+        if (!$beforeNeedle) {
+
+            if (false === $part = strrchr($this->string, $needle)) {
+                return;
+            }
+
+            $result = clone $this;
+            $result->string = $part;
+
+            return $result;
+        }
+
+        if (false === $offset = strrpos($this->string, $needle)) {
+            return;
+        }
+
+        $result = clone $this;
+        $result->string = substr($this->string, 0, $offset);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function lastSubstringOfIgnoreCase(string $needle, bool $beforeNeedle = false)
+    {
+        if ('' === $needle) {
+            return;
+        }
+
+        if (false === $offset = strripos($this->string, $needle)) {
+            return;
+        }
+
+        $result = clone $this;
+
+        $current = setlocale(LC_CTYPE, '0');
+        setlocale(LC_CTYPE, 'C');
+        $result->string = $beforeNeedle ? substr($this->string, 0, $offset) : substr($this->string, $offset);
+        setlocale(LC_CTYPE, $current);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reverse()
+    {
+        $result = clone $this;
+        $result->string = strrev($this->string);
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function width(bool $ignoreAnsiDecoration = true): int
+    {
+        $s = $this->string;
+
+        if (false !== strpos($s, "\r")) {
+            $s = str_replace("\r\n", "\n", $s);
+            $s = strtr($s, "\r", "\n");
+        }
+
+        $width = 0;
+        foreach (explode("\n", $s) as $s) {
+            if ($ignoreAnsiDecoration) {
+                $s = preg_replace('/\x1B\[[\d;]*m/', '', $s);
+            }
+
+            if ($width < $c = strlen($s)) {
+                $width = $c;
+            }
+        }
+
+        return $width;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function replace($from, $to, int &$count = null)
+    {
+        if (!is_string($from) && !is_array($from)) {
+            throw new InvalidArgumentException('Search pattern must be a valid string or array of strings.');
+        }
+
+        if (!is_string($to) && !is_array($to)) {
+            throw new InvalidArgumentException('Pattern replacement must be a valid string or array of strings.');
+        }
```

Comment:
These checks are not proper and might still lead to an array to string conversion, because thhe second argument can be an array only if the first argument is an array.